src/main/java/edu/harvard/iq/dataverse/engine/command/impl/ReturnDatasetToAuthorCommand.java

```diff
@@ -0,0 +1,97 @@
+package edu.harvard.iq.dataverse.engine.command.impl;
+
+import edu.harvard.iq.dataverse.Dataset;
+import edu.harvard.iq.dataverse.DatasetVersionUser;
+import edu.harvard.iq.dataverse.DvObject;
+import edu.harvard.iq.dataverse.UserNotification;
+import edu.harvard.iq.dataverse.authorization.Permission;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.engine.command.AbstractCommand;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.engine.command.exception.IllegalCommandException;
+import edu.harvard.iq.dataverse.util.BundleUtil;
+import edu.harvard.iq.dataverse.workflows.WorkflowAction;
+import static edu.harvard.iq.dataverse.workflows.WorkflowAction.Type.RETURN_TO_AUTHOR;
+import edu.harvard.iq.dataverse.workflows.WorkflowComment;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.Future;
+
+@RequiredPermissions(Permission.PublishDataset)
+public class ReturnDatasetToAuthorCommand extends AbstractCommand<Dataset> {
+
+    private final Dataset theDataset;
+    private final String comment;
+
+    public ReturnDatasetToAuthorCommand(DataverseRequest aRequest, DvObject anAffectedDvObject, String comment) {
+        super(aRequest, anAffectedDvObject);
+        this.theDataset = (Dataset) anAffectedDvObject;
```

Comment:
why not force anAffectedDvObject to be a dataset, as the conversion is forced here?